lib/active_job_status/job_tracker.rb

```diff
@@ -30,28 +30,27 @@ def performing
         job_id,
         JobStatus::WORKING.to_s,
         expires_in: expiration || DEFAULT_EXPIRATION
-
       )
     end
 
     def completed
+      previous_status = store.fetch(job_id)
```

Comment:
have to fetch this before we change it otherwise maybe_remove_from_batch will never remove from batch